zebra-state/tests/basic.rs

```diff
@@ -38,13 +42,13 @@ static GET_TIP_TRANSCRIPT: Lazy<Vec<(Request, Response)>> = Lazy::new(|| {
         // Insert higher block first, lower block second
         (
             Request::AddBlock { block: block1 },
-            Response::Added { hash: hash1 },
+            Ok(Response::Added { hash: hash1 }),
         ),
         (
             Request::AddBlock { block: block0 },
-            Response::Added { hash: hash0 },
+            Ok(Response::Added { hash: hash0 }),
         ),
-        (Request::GetTip, Response::Tip { hash: hash1 }),
+        (Request::GetTip, Ok(Response::Tip { hash: hash1 })),
```

Comment:
I think it would be good to extend this transcript with an error case to see the error ergonomics in practice.

Reply:
I didn't do it for this example but I did add an example using the fn to the `zebra-tests/tests/transcript.rs` file

Reply:
Ended up doing that here https://github.com/ZcashFoundation/zebra/pull/678/files#diff-9bc388fb6fb0b6f6cadd80bbd760b1c7R57